src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java

```diff
@@ -20,7 +20,7 @@ public CompositeIdFetcher(ImportFormatPreferences importFormatPreferences) {
     }
 
     public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
-        Optional<DOI> doi = DOI.parse(identifier);
+        Optional<DOI> doi = DOI.parse(filterForDOITag(identifier));
```

Comment:
Please no new method if not needed.

Please to 

```java
Optional<DOI> doi = DOI.findInText(identifier);
```

and report back if this works.

Reply:
That certainly works. Sorry, I misunderstood the requirements you originally outlined:

> Test-enabling: Heuristics as own method. This method needs to be tested. Method can be "package private".

Reply:
Nice!

Sorry for having a discussion with you without updating the main text. I thought, it was clear.

4 days ago, we discussed the usage of that method: https://github.com/JabRef/jabref/issues/9821#issuecomment-1533253820

(And you even acknowledged 4 days ago - https://github.com/JabRef/jabref/issues/9821#issuecomment-1533588857)

Sorry again that I did not update the main text.

Reply:
Ah, ok. No worries, my misunderstanding. It's really quite a small fix - I would have been quicker except it didn't occur to me for most of yesterday that the reason it wasn't working was the copy-paste issue.

Reply:
The copy and paste thing is a hard one to work on. This involves understanding JavaFX behavior on macOS, which is different than on Windows. Therefore, we have some "hacks" included, which might not work as expected (which you encountered).

Nevertheless, the issue at hand requires some GUI (!) tests, which could be hard. See https://github.com/JabRef/jabref/blob/main/src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java for an example test.

Side story: It is hard to craft "medium-sized" issues. Mostly, the issues involve much code understanding, but little code changes. -- You can look at https://github.com/orgs/JabRef/projects/3/views/3 for a curated list of issues...